src/services/yolo_detection_network.py

```diff
@@ -41,9 +66,45 @@ def validate(cls, config: ServiceConfig) -> Sequence[str]:
     def reconfigure(
         self, config: ServiceConfig, dependencies: Mapping[ResourceName, ResourceBase]
     ):
-        self.cfg = YDNConfig(config.attributes.fields)
+        """
+        This reconfigure method reinitializes the config and updates the YDNConfig
+        on the component pipeline side using a do_command.
+
+        Args:
+            config (ServiceConfig): Viam service config
+            dependencies (Mapping[ResourceName, ResourceBase]): list of deps
+
+        Raises:
+            e: KeyError if camera dep is not found in dependencies
+        """
+        self.cfg = YDNConfig(config.attributes.fields, config.name)
         self.cfg.initialize_config()
-        LOGGER.info(f"Native YDN config: {vars(self.cfg)}")
+
+        self.cam_name = config.attributes.fields["cam_name"].string_value
+        try:
+            self.cam = dependencies[Camera.get_resource_name(self.cam_name)]
+        except KeyError as e:
+            self.logger.error(
+                f"Could not find camera dep '{self.cam_name}' in deps list."
+            )
+            raise e
+        self.ydn_service_name = config.name
+
+        async def async_do_command():
+            while not self.pipeline_configured and self.should_exec:
```

Comment:
[q] Is it worth adding a max attempts check here in case conigure_cmd keeps erroring out?

Reply:
Added a max of 5